server/WAPI.rb

```diff
@@ -252,6 +272,31 @@ def get_request(request)
         wrapped_response = do_request(request)
       end
     end
+
+    # If it didn't work just return that information.
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: wrapped_response: meta_status: #{wrapped_response.meta_status}"
+    if wrapped_response.meta_status != WAPI::SUCCESS
+      return wrapped_response
+    end
+
+    # Ran a query successfully.  See if got partial data and need to keep going.
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: wrapped_response: data length: #{wrapped_response.result.length}"
+
+    # See if there is a link header, if so get the rest of the data.
+    if wrapped_response.meta_more.length > 0
+      logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: found link header: >>#{wrapped_response.meta_more}<<"
+
+      more_data = get_request(wrapped_response.meta_more)
```

Comment:
Wonder which way is better: to do the recursive call to get all pages, or do a loop given that the total page count is returned in the first call result.